ckanext/opendatani/plugin.py

```diff
@@ -208,6 +219,34 @@ def report_resources_by_organization(context, data_dict):
     return resource_data
 
 
+@toolkit.side_effect_free
+def prepare_csv_report(context, data_dict):
+    """Creates csv file and stores it under CKAN's storage path.
+    :return: a string containing the csv_id of the created archive
+    :rtype: string
+    """
+    file_name = uuid.uuid4().hex + '.{ext}'.format(ext='csv')
+    file_path = helpers.get_storage_path_for('temp-ni') + '/' + file_name
+
+    try:
+        with open(file_path, 'w') as csvfile:
+            resource = toolkit.get_action('report_resources_by_organization')(context, data_dict)
+            fields = resource[0].keys()
```

Comment:
What if `resource` is empty? We should consider that case as well as there might not be datasets for the organization at all (that is a really dumb case but still)

Generally, think this function should not bother about acquiring the data. It should accept the list of the data as an argument and put it in the file (and return the data). All the rest should be done by `toolkit.get_action('report_resources_by_organization')(context, data_dict)`